### PR TITLE
Revert "remove guava deps from core"

### PR DESCRIPTION
**Pull request by mattnworb**
Reverts spotify/missinglink#56

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -6,7 +6,7 @@ secure: "U9dY+CGViIQmq/vU6t0kByXZHQnwi5b+vhNUdjU40XzKKgJRa/WV4us7wNkICljDAFv1lto
 language: java
 
 jdk:
-  - openjdk8
+  - oraclejdk8
```
Comment (spkrka) [inline .travis.yml]: This part shouldn't be reverted

File: core/pom.xml [hunk 2]
```diff
@@ -28,6 +28,11 @@
       <version>0.11.0</version>
       <scope>provided</scope>
     </dependency>
+    <dependency>
+      <groupId>com.google.guava</groupId>
+      <artifactId>guava</artifactId>
+      <version>18.0</version>
```
Comment (spkrka) [inline core/pom.xml]: If we get it transitively it's not a huge issue I suppose

File: pom.xml [hunk 3]
```diff
@@ -153,7 +153,7 @@
           <dependency>
             <groupId>com.puppycrawl.tools</groupId>
             <artifactId>checkstyle</artifactId>
-            <version>8.20</version>
+            <version>[8.20,)</version>
```
Comment (spkrka) [inline pom.xml]: Don't revert this

File: core/pom.xml (hunk 2 above)
Comment (mattnworb) [reply core/pom.xml]: No, but rather avoid surprises for now - plus I think guava comes in transitively at `provided` currently

**Comment by mattnworb** [comment]
@spkrka d'oh, I re-did this to revert just the one commit than the whole PR.

**Pull request merged**